src/Umbraco.Web/PublishedContentQuery.cs

```diff
@@ -191,46 +191,52 @@ public IEnumerable<PublishedSearchResult> Search(string term, string culture = "
         /// <inheritdoc />
         public IEnumerable<PublishedSearchResult> Search(string term, int skip, int take, out long totalRecords, string culture = "*", string indexName = null)
         {
-            indexName = string.IsNullOrEmpty(indexName)
-                ? Constants.UmbracoIndexes.ExternalIndexName
-                : indexName;
+            if (string.IsNullOrEmpty(indexName))
+            {
+                indexName = Constants.UmbracoIndexes.ExternalIndexName;
+            }
 
             if (!_examineManager.TryGetIndex(indexName, out var index) || !(index is IUmbracoIndex umbIndex))
+            {
                 throw new InvalidOperationException($"No index found by name {indexName} or is not of type {typeof(IUmbracoIndex)}");
+            }
 
             var searcher = umbIndex.GetSearcher();
 
-            // default to max 500 results
-            var count = skip == 0 && take == 0 ? 500 : skip + take;
```

Comment:
There's no need to also have a hardcoded `maxResults` here, so I've changed it to be the same as the overload below.